File: 23_12_week2/wangon/11497.py
from collections import deque

t = int(input())
for _ in range(t):
    n = int(input())
    ret = -1
    arr = list(map(int, input().split()))
    d = deque()
    arr.sort()
    d.append(arr[-1])
    arr.pop()
    flag = 1
    while len(arr):
        if flag == 1:
            d.append(arr[-1])
        else:
            d.appendleft(arr[-1])
        flag *= -1
        arr.pop()
    for i in range(n-1):
        ret = max(ret, abs(d[i] - d[i+1]))
    ret = max(ret, abs(d[-1] - d[0]))
    print(ret)

File: src/23_12_week2/wangon/11497.cpp
#include <algorithm>
#include <cstdlib>
#include <iostream>
#include <string>
#include <vector>
#include <deque>
using namespace std;

int main()
{
	int t;
	cin >> t;
	while (t--)
	{
		int n;
		cin >> n;
		int ret = -1;
		vector<int> arr(n);
		deque<int> tmp;
		for (int i = 0; i < n; i++)
			cin >> arr[i];
		sort(arr.begin(), arr.end());
		tmp.push_back(*(arr.end() - 1));
		arr.pop_back();
		int flag = 1;
		while (arr.size() > 0)
		{
			if (flag == 1)
				tmp.push_back(*(arr.end() - 1));
			else
				tmp.push_front(*(arr.end() - 1));
			flag *= -1;
			arr.pop_back();
		}
		for (int i = 0; i < n - 1; i++)
			ret = max(ret, abs(tmp[i] - tmp[i+1]));
		ret = max(ret, abs(tmp[n-1] - tmp[0]));
		cout << ret << endl;
	}
}
